modules/materialize-instance/main.tf

```diff
@@ -0,0 +1,103 @@
+locals {
+  instance_namespace = var.instance_namespace != null ? var.instance_namespace : var.operator_namespace
```

Comment:
If we're already breaking backwards compatibility, just require `var.instance_namespace`.